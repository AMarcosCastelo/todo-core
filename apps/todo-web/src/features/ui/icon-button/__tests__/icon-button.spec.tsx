import { Check } from 'lucide-react';

import { baseRender } from '~/utils/testing';

import IconButton from '../icon-button';

describe('Todo form', () => {
  const defaultProps = {
    ariaLabel: 'Check Button',
    icon: <Check data-testid="check-icon" />,
  };
  const component = (customProps = {}) => (
    <IconButton {...customProps} {...defaultProps} />
  );

  it('should render correctly', () => {
    const screen = baseRender(component());

    expect(screen.container).toMatchSnapshot();
  });

  it('should render the icon', () => {
    const screen = baseRender(component());

    expect(screen.getByTestId('check-icon')).toBeInTheDocument();
  });
});
