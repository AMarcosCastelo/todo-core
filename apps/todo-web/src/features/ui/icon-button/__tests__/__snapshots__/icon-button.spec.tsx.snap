// Vitest Snapshot v1

exports[`Todo form > should render correctly 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <button
      aria-label="Check Button"
      class="nextui-c-eWzLKI"
    >
      <svg
        class="lucide lucide-check"
        data-testid="check-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </button>
  </div>
</div>
`;
